Conclusions

agitter opened the pull request:
@cgreene see if you agree with the position I took, and feel welcome to tear down this entire section if you have a different stance.

This will be one focal point of the review, so others can provide feedback as well.

enricoferrero commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+its dominance over competing machine learning approaches in many of the areas
+reviewed here and quantitative improvements in predictive performance, deep
+learning has not yet qualitatively "solved" those problems that were previously
+"unsolved".
```
Is 'that were previously "unsolved" needed here? It sounds like a bit of a repetition to me.

agitter replied:
Cut text

enricoferrero commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+finally approaching or exceeding human performance in the past year
+[@arxiv:1610.05256 @arxiv:1703.02136] `TODO: working on a second source for this
+error trajectory from a talk by Eric Horvitz`. The phenomenal  improvements on
+benchmark datasets are undeniable, but the successes of the early 2010s did not
```
Maybe it's just me but I don't think it's overly clear what the 

> successes of the early 2010s

 are.

agitter replied:
Good point, this is unclear.  I'm thinking of the left image from https://twitter.com/amram/status/845748240033050624  However, I can't directly use it because I don't know the source, and Eric is likely too busy being the new head of MSR to reply.  http://www.businessinsider.com/ibm-edges-closer-to-human-speech-recognition-2017-3 shows comparable numbers but with less granularity.

I changed this to "...are undeniable, but halving the error rates on these benchmarks did not fundamentally transform...".  Still not perfect, but that's my intended message.

No one is pushing back on my conversational speech example, so I removed: `TODO: this is debatable, maybe need a different example or to clarify what is meant by "conversational" speech`

enricoferrero commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+fundamentally transform the domain. `TODO: this is debatable, maybe need a
+different example or to clarify what is meant by "conversational" speech`
+Widespread adoption of these technologies will requires not only improvements
+over baseline methods but truly "solving" the problem, in this case exceeding
```
Since this is the second time you use 'solve' in this section I think the double quotes are unnecessary.

agitter replied:
Fixed

enricoferrero commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+[@doi:10.1001/jama.2016.17216], diabetic macular edema
+[@doi:10.1001/jama.2016.17216], and skin lesion [@doi:10.1038/nature21056]
+classifiers are highly accurate and comparable to dermatologist performance in
+the latter case. `TODO: more imaging examples or other examples that might be at
```
> comparable to dermatologist performance in the latter case
sounds too specific for this section, maybe consider something more generic like
> comparable to human performance

enricoferrero replied:
Agree, maybe you can add #366 here?

agapow replied:
Agreed - reads easier.

agitter replied:
"comparable to clinician performance" to differentiate between an expert and average human

agitter replied:
Added it.  I like the idea about using experts to review a smaller set of images where two CNNs disagree, but that may be better for the Categorize sub-section instead of expanding the conclusions.

enricoferrero commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+[@doi:10.1001/jama.2016.17216], and skin lesion [@doi:10.1038/nature21056]
+classifiers are highly accurate and comparable to dermatologist performance in
+the latter case. `TODO: more imaging examples or other examples that might be at
+or close to "transformative"?`  In other domains, perfect accuracy will not be
```
Do we want to argue that in some domains perfect accuracy might not even be possible? Not sure what arguments we could bring here but intuitively some complex problems (e.g: gene regulation in disease) might not be within DL reach, no matter how good the algorithm is.

Just a thought, I appreciate it's a rather subjective stance and not everyone might agree.

cgreene replied:
I agree with @enricoferrero that in some areas perfect performance may be impossible for any algorithm due to the stochastic/critical nature of biological systems. Maybe just one closing sentence to this paragraph since it's the perfect accuracy may not... paragraph.  Also, I think you may want to start a new paragraph at "In other domains...".

@agitter : if you want to create an issue for this, I or @enricoferrero can likely address afterwards. Up to you on what you prefer.

agitter replied:
I completely agree with this sentiment.  I reworked this slightly; see if it needs more editing and a new issue.  My intention is to keep the two examples where deep learning is closer to practical impact (medical imaging and chemical screening) together.   Then I split a new paragraph for the more negative examples.

enricoferrero commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+approaches that would be infeasible with other machine learning techniques.
+Unsupervised methods are currently less-developed than their supervised
+counterparts, making them an attractive target for future research in this
+domain.  `TODO: still working on a strong closing line`
```
I feel this sentence on unsupervised learning could be expanded a little bit. Maybe we could argue that since a lot of biomedical data is unlabelled and labelling has to be done manually, accurate deep unsupervised methods could also be transformative in high impact fields such as patient stratification for precision medicine approaches.

agitter commented:
@enricoferrero thanks for these comments.  These are excellent suggestions, and I'll make point-by-point responses and text updates, hopefully tonight.

cgreene commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+For example, in chemical screening for drug discovery, a deep learning system
+that successfully identifies dozens or hundreds of target-specific, active
+small molecules from a massive search space would have immense practical value
+even if its overall precision is modest. Conversely, the most challenging tasks
```
I would need some clarification here to decide if this should be elaborated on. Why are these potentially the most challenging? I guess I am missing the rationale for this statement.

agitter replied:
Perhaps I mean to say that errors may be magnified in some cases if we rely on predictions for some secondary task.  For instance, suppose I rely on predictions of TF binding from a neural net to predict TF binding and then use those genome-wide predictions to model gene expression.  I'll need to be more accurate than if I were going to use those predictions to follow up on a few interesting or high-confidence binding sites.

cgreene commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+challenges beyond improving training and predictive accuracy, such as preserving
+patient privacy and interpreting models.  Ongoing research has begun to address
+these problems and shown they are not insurmountable.  Deep learning offers the
+flexibility to model data in its most natural form, spurring creative modeling
```
What do you mean by "most natural form" here?

agitter replied:
I'm thinking of things like graph convolutional networks that allow one to work directly on a molecular graph for chemical modeling.  Previously, pre-computed lossy feature representations are more common when using other machine learning approaches.   I could probably come up with examples in other domains as well.

cgreene commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+these problems and shown they are not insurmountable.  Deep learning offers the
+flexibility to model data in its most natural form, spurring creative modeling
+approaches that would be infeasible with other machine learning techniques.
+Unsupervised methods are currently less-developed than their supervised
```
Unsupervised methods are currently less-developed than their supervised counterparts, but they may have the most potential. When deep learning algorithms can summarize very large collections of input data into interpretable models that spur scientists to ask questions that we didn't previously know to ask, it will be clear that deep learning has transformed biology and medicine.

agitter replied:
Done

agitter commented:
@cgreene thanks, I'll be able to clarify these parts in my next pass.

agapow commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+its transformative potential or induced a strategic inflection point.  Despite
+its dominance over competing machine learning approaches in many of the areas
+reviewed here and quantitative improvements in predictive performance, deep
+learning has not yet qualitatively "solved" those problems that were previously
```
instead of "qualitatively" maybe "definitively" / "inarguably"

agitter replied:
Changed

agapow commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+human-level performance, as well as convincing users to embrace the technology
+[@tag:Speech_recognition].  We see parallels to the healthcare domain, where
+achieving the full potential of deep learning will require outstanding
+predictive performance as well as adoption by biologists and clinicians.
```
maybe "acceptance and adoption"

agitter replied:
Changed

agapow commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+classifiers are highly accurate and comparable to dermatologist performance in
+the latter case. `TODO: more imaging examples or other examples that might be at
+or close to "transformative"?`  In other domains, perfect accuracy will not be
+required because deep learning will be used primarily to prioritize experiments.
```
I like this point a lot: imperfect solutions can be damn useful. Perhaps it needs to be broken out to a separate paragraph to emphasis? I'm thinking there's a lot of value in "assisted discovery" and highlighting areas for further investigation. Conversely, this could blow out into far too long a discussion.

agitter replied:
Glad you like it.  I made a separate paragraph and added an example from #366.  We could probably add at least one more to make the point stronger if you have ideas.

agitter replied:
I should add, I'm somewhat concerned the false negatives in the deep learning-assisted strategy proposed in #366.

agapow commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
+decision-making, especially in the clinic. `TODO: elaborate more on this idea
+or split in a new paragraph?`
+
+Even if deep learning in biology and healthcare is not yet transformative today,
```
... and it's early days yet, full potential not explored, deep learning still evolving, yadda yadda yadda

agitter replied:
Added a line

agitter commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,60 @@
 
-Final thoughts and future outlook here. The Discussion will give an overview
-and the Conclusion will provide a short, punchy take home message.
+Deep learning-based methods now represent the state of the art in a diverse
```
Yes, that's better

agitter commented:
Thanks again @enricoferrero @cgreene @agapow.  I updated the text or commented in response to all feedback above.  There are still some open questions we can resolve before merging, mostly in response to @cgreene.

agapow commented on sections/07_conclusions.md:
```diff
@@ -1,60 +1,70 @@
-"unsolved".
+Deep learning-based methods now matches or surpasses the previous state of the
+art in a diverse array of tasks in patient and disease categorization,
+fundamental biological study, genomics, and treatment development.  We return to
```
"Returning to our central question": less passive

agitter replied:
Yes, that's better.  I changed it.

cgreene commented on sections/07_conclusions.md:
```diff
@@ -1,15 +1,70 @@
 
-Final thoughts and future outlook here. The Discussion will give an overview
-and the Conclusion will provide a short, punchy take home message.
+Deep learning-based methods now matches or surpasses the previous state of the
```
subject/verb agreement

agitter commented:
Made the last change from @cgreene and addressed a couple TODOs.  Merging now.

I added this example to clarify one of my remarks: "As an example, errors in a predicted protein contact map could be amplified if that contact map is used directly for 3D structure prediction." @j3xugit, is this statement correct in your opinion?

j3xugit commented:
In fact, we can directly use predicted contact maps to predict 3D
structures and for many proteins we indeed produce very good 3D modeling.
The prediction error of an individual contact may be big, but when multiple
predicted contacts are used together for 3D structure modeling, the impact
of an individual contact is usually reduced instead of amplified.

By the way, I believe that the change of ab initio folding is
transformative in the past few years due to 1) significantly improved
co-evolution analysis; and 2) significantly improved contact prediction by
deep learning.

On Thu, May 4, 2017 at 9:16 AM, Anthony Gitter <notifications@github.com>
wrote:

> Made the last change from @cgreene <https://github.com/cgreene> and
> addressed a couple TODOs. Merging now.
>
> I added this example to clarify one of my remarks: "As an example, errors
> in a predicted protein contact map could be amplified if that contact map
> is used directly for 3D structure prediction." @j3xugit
> <https://github.com/j3xugit>, is this statement correct in your opinion?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/greenelab/deep-review/pull/370#issuecomment-299197967>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKR63opN4qoO1559KyTxbWAMZQ2wZR0oks5r2d2mgaJpZM4NNtGm>
> .
>



-- 
_________________________________________
Professor
Toyota Technological Institute at Chicago
6045 S. Kenwood Ave.
Chicago, IL 60637
fax: 773 834 2557, Google Voice: 773 359 3721
http://ttic.uchicago.edu/~jinbo/

agitter commented:
@j3xugit thanks for the correction.  I opened #376 to address this and will make a new pull request.
